internal/processing/workers/fromfediapi_move.go

```
@@ -0,0 +1,459 @@
+		// to ensure we have up-to-date version.
+		targetAcct, _, err = p.federate.RefreshAccount(ctx,
+			receiver.Username,
+			targetAcct,
+			targetAcctable,
+			dereferencing.Freshest,
+		)
+		if err != nil {
+			return gtserror.Newf(
+				"error refreshing target account %s: %w",
+				targetAcctURIStr, err,
+			)
+		}
+	}
+
+	/*
+		At this point we have an up-to-date version
+		of the Move origin account (not ours), and
+		also the Move target account (possibly ours).
+	*/
+
+	// Target must not itself have moved somewhere.
+	// You can't move to an already-moved account.
+	targetAcctMovedTo := targetAcct.MovedToURI
+	if targetAcctMovedTo != "" {
+		l.Infof(
+			"target account has, itself, already moved to %s, will not process Move",
+			targetAcctMovedTo,
+		)
+		return nil
+	}
+
+	// Target must be aliased back to origin account.
+	// Ie., its alsoKnownAs values must include the
+	// origin account, so we know it's for real.
+	if !targetAcct.IsAliasedTo(originAcctURIStr) {
+		l.Info("target account is not aliased back to origin account, will not process Move")
+		return nil
+	}
+
+	/*
+		At this point we know that the move
+		looks valid and we should process it.
+	*/
+
+	var errs gtserror.MultiError
+
+	// Transfer originAcct's followers
+	// on this instance to targetAcct.
+	if err := p.RedirectAccountFollowers(
+		ctx,
+		originAcct,
+		targetAcct,
+	); err != nil {
+		errs.Append(err)
+	}
+
+	// Remove follows on this
+	// instance owned by originAcct.
+	if err := p.RemoveAccountFollowing(
+		ctx,
+		originAcct,
+		targetAcct,
+	); err != nil {
+		errs.Append(err)
+	}
+
+	// Whatever happened above, error or
+	// not, we've just at least attempted
+	// the Move so we'll need to update it.
+	move.AttemptedAt = time.Now()
+	updateColumns := []string{"attempted_at"}
+
+	if err := errs.Combine(); err != nil {
+		// We tried to process but
+		// didn't succeed completely.
+		l.Infof("one or more errors procesing Move side effects: %v", err)
```

did you mean to put error level? :p

---

though thinking about it, i think it might be a bit easier to read in the logs if instead of using a multi error you instead just put a `l.Errorf("error .... : %v", err)`in each of the error checking clauses above so they're separate log entries

---

Alright, I changed the error handling (here and elsewhere) so the called functions log the error and just return a `true` bool if it went OK, since we don't need to bubble the errors up anywhere or store them anywhere here.